src/extension.ts

```diff
@@ -207,17 +207,45 @@ async function autoUpdate() {
     }
 }
 
-function diagnosticCounter() {
+function progressCounter() {
+    const runningProgress: { [index: string]: boolean } = {};
```

Comment:
I'm really wonky with JS, but would it be possible to replace that with [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) for clarity?

Also not really sure why/how it's allowed to be `const`, since it's clearly modified in the callbacks, but I guess that's TypeScript shenanigans that I'm missing.

Reply:
Agreed. `Set` is better.

`const` is not typescript shenanigans but javascript. It's not actually immutable (sadly), it's just the actual variable itself that is impossible to change. I.e.

```
const foo = 42;
foo = 43; // FAIL
```

Reply:
Right, makes sense. Thanks for the changes!